rgw/v2/tests/s3_swift/reusable.py

```diff
@@ -1010,6 +1010,10 @@ def check_datalog_marker():
             get_datalog_marker = datalog_status[i]["marker"]
             break
 
+    # return shard_id and datalog_mark, Ref BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1981860
+    if get:
+        return i, get_datalog_marker
+
```

Comment:
IMO, we could write a separate method to retrieve the marker instead of combining it with the check method. It could cause confusion.

Here, `i` is out of scope and may not be reliable though it would retain the last value. However, there is a possibility that we can traverse the entire range and return an incorrect `shard_id`

Reply:
Sure @psathyan made the changes